src/networking.c

```diff
@@ -103,7 +103,8 @@ client *createClient(connection *conn) {
     }
 
     selectDb(c,0);
-    uint64_t client_id = ++server.next_client_id;
+    uint64_t client_id;
+    atomicGetIncr(server.next_client_id, client_id, 1);
```

Comment:
Looks like there's an off-by-one here, `atomicGetIncr()` seems to implement `val++` rather than `++val`.

Reply:
Yes, `server.next_client_id` starts from 1, is there a problem? and It was like this before C11 _Atomic

Reply:
The first initialized client (`server.lua_client`) would be assigned with `client_id` of `1` instead of `2` which was the case until now (because `next_client_id` is initialized to `1` and not `0`).

I can't think of a problem this would introduce (maybe it would if `next_client_id` was initialized to zero), but wanted to point out and be aware of this side effect.

Reply:
Yes, `next_client_id` is initialized to 1, just as the name means, it represents the "next" id.

Reply:
the original code actually look like an unintended bug (they client IDs started from 2).
```c
    server.next_client_id = 1; /* Client IDs, start from 1 .*/
...
    uint64_t client_id = ++server.next_client_id;
```
and i agree that the name suggests this is the **next** client id, so although this may be an unintended side effect, it's actually somewhat of a bugfix.